Packs/Slack/Integrations/Slack/Slack.py

```diff
@@ -1844,6 +1852,28 @@ def init_globals():
     BOT_ICON_URL = demisto.params().get('bot_icon')
     MAX_LIMIT_TIME = int(demisto.params().get('max_limit_time', '60'))
     PAGINATED_COUNT = int(demisto.params().get('paginated_count', '200'))
+    SLACK_COMMAND_TIMEOUT = int(demisto.params().get('timeout', 2))
+
+
+class SlackCommandTimedThread(threading.Thread):
+    """
+        Attributes:
+            target (method): The method the created thread will run.
+            timeout (int): how many seconds does the thread have untill it is terminated
+    """
+    def __init__(self, target, timeout=2):
+        self.timeout = timeout
+        threading.Thread.__init__(self, target=target)
+
+    def timed_run(self):
+        """Run the thread for the given time.
+
+        Returns:
+
+        """
+        self.start()
+        time.sleep(self.timeout)
+        self.join()
```

Comment:
where are you exiting the thread?

Reply:
no need apperantly - I'll update the description